src/lamplib/src/genny/curator.py

```diff
@@ -145,45 +146,82 @@ def poplar_grpc(cleanup_metrics: bool, workspace_root: str, genny_repo_root: str
     prior_cwd = os.getcwd()
     try:
         os.chdir(workspace_root)
-        poplar = subprocess.Popen(args)
-        if poplar.poll() is not None:
-            raise OSError("Failed to start Poplar.")
-        try:
-            os.chdir(prior_cwd)
-            connected = False
-            for i in range(10):
-                time.sleep(0.2)  # sleep to let curator get started. This is a heuristic.
-                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-                try:
-                    sock.connect(("127.0.0.1", 2288))
-                    # If we didn't throw an exception, then we successfully connected
-                    connected = True
-                    break
-                except socket.error as socket_error:
-                    if socket_error.errno != errno.ECONNREFUSED:
-                        # We expect to get connection refused if poplar is not up yet.
-                        # If we get something else, bail
-                        raise socket_error
-                finally:
-                    sock.close()
-            if not connected:
-                raise OSError(f"Poplar not listening on port 2288")
-            yield poplar
-        finally:
+        with open("./build/WorkloadOutput/poplar_grpc.log", "w+") as log_file_obj:
```

Comment:
Where did this filename come from?

Reply:
I chose this file name because the enclosing function is named `poplar_grpc`

Reply:
Sorry, I should've been more specific, is `./build/WorkloadOutput` a magic basedir name that makes this meaningful? I ask because I think evergreen very recently added "pick up logs from a directory" and I was wondering whether this was to enable that, or whether this is a required directory etc.

Reply:
My understanding is: Genny writes test results, metrics etc. into this directory; DSI downloads the content of this directory during `make_artifacts`. Since this log file maybe needed for debugging, I put it into this directory.